src/meta/client/MetaClient.cpp

```diff
@@ -1123,6 +1114,31 @@ StatusOr<SchemaVer> MetaClient::getNewestEdgeVerFromCache(const GraphSpaceID& sp
 }
 
 
+bool MetaClient::diffLeader(const LeaderIds& oldLeaderIds, const LeaderIds& newLeaderIds) {
+    for (const auto& dist : newLeaderIds) {
+        auto spaceIt = oldLeaderIds.find(dist.first);
+        if (spaceIt == oldLeaderIds.end()) {
+            return true;
+        } else {
+            if (dist.second.size() != spaceIt->second.size()) {
+                return true;
+            } else {
+                for (const auto& partId : dist.second) {
+                    auto it = std::find_if(spaceIt->second.begin(), spaceIt->second.end(),
+                            [&] (const auto& item) {
+                                return item == partId;
+                        });
+                    if (it == spaceIt->second.end()) {
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+    return false;
+}
+
```

Comment:
I have a question，Is the above code equal to this:
```
bool MetaClient::diffLeader(const LeaderIds& oldLeaderIds, const LeaderIds& newLeaderIds) {
      return (oldLeaderIds != newLeaderIds);
  }
```

Reply:
using LeaderIds = std::unordered_map<GraphSpaceID, std::vector<PartitionID>>;
so, the answer maybe `No`.

Reply:
Wow , Just a discussion, I'm not sure it's appropriate at here. :smiley_cat:

```
    typedef std::unordered_map<int64_t, std::vector<int64_t>> map;
    map a = { {1, {1,2,3}}, {2, {1,2,3}}, {3, {1,2,3}}};
    map b = { {3, {1,2,3}}, {2, {1,2,3}}, {1, {1,2,3}} };
    map c = { {3, {2,3}}, {2, {1,2,3}}, {1, {1,2,3}} };

  if (a==b) {
      LOG(INFO) << "a and b are equal";    <--- hit
  }

  if (a!=c) {
      LOG(INFO) << "a and c are not equal";     <--- hit
  }
```

Reply:
FYI. https://en.cppreference.com/w/cpp/container/unordered_map/operator_cmp
You are right! Then, I have learnt the `unordered_map` and cleared some useless code in the PR about it.
Thx.

Reply:
👍